packages/fxa-auth-server/lib/devices.js

```diff
@@ -117,7 +116,7 @@ module.exports = (log, db, push) => {
           // so try to synthesize one if necessary.
           let deviceName = result.name;
           if (! deviceName) {
-            deviceName = synthesizeName(deviceInfo);
+            deviceName = credentials.client && credentials.client.name || synthesizeName(deviceInfo);
```

Comment:
I'm trying to do more "fill in defaults on read" than "fill in defaults on write", so that e.g. if we update the oauth client name it will affect all corresponding device records. (The code that sets the name on write is still in place though, because I couldn't make it work nicely in all cases).